### PR TITLE
Fix user validation for some fields

### LINKED ISSUE #136: [PE-D][Tester D] Numbers should not be unlimited
Phone number input seem to accept extremely long and unrealistic numbers.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: tjch-o/ped#4

### LINKED ISSUE #138: [PE-D][Tester D] Contact name should not be numbers
It does not make sense to have names that can accept numbers.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: tjch-o/ped#5

### LINKED ISSUE #153: [PE-D][Tester B] GPA field can accept more than 2 decimal places
In the User Guide, GPA can take on numbers with 0, 1 or 2 decimal places, but I was able to add numbers with more than 2 decimal places

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: darrentfy/ped#4

### LINKED ISSUE #167: [PE-D][Tester C] GPA can be more than 2dp
When I ran 'edit 2 g/3.45567899':

But the prefix summary in UG says:

This could be misleading.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: nananakx-x/ped#22

### PR DESCRIPTION
**Pull request by nabonitasen**
Closes #136, closes #138, closes #153, closes #167.

This PR fixes bugs that allow incorrect inputs into fields.

### CODE REVIEW

**Review by ravern — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/person/Gpa.java [hunk 1]
```diff
@@ -9,7 +9,8 @@
  */
 public class Gpa implements Comparable<Gpa> {
 
-    public static final String MESSAGE_CONSTRAINTS = "Gpa can take values from 0.00 to 5.00, and cannot be blank";
+    public static final String MESSAGE_CONSTRAINTS = "Gpa can take values from 0.00 to 5.00 up to 2 decimal places, "
+            + "and cannot be blank";
```
Comment (amosting) [inline src/main/java/seedu/address/model/person/Gpa.java:13]: Add a fullstop

File: src/main/java/seedu/address/model/person/Name.java [hunk 2]
```diff
@@ -10,13 +10,14 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain alphanumerical characters and spaces with an alphabetical first name, "
+                    + "and it should not be blank";
```
Comment (amosting) [inline src/main/java/seedu/address/model/person/Name.java:14]: add a fullstop

File: src/main/java/seedu/address/model/person/Phone.java [hunk 3]
```diff
@@ -11,8 +11,8 @@ public class Phone {
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Phone numbers should only contain numbers, and it should be at least 3 digits long";
-    public static final String VALIDATION_REGEX = "\\d{3,}";
+            "Phone numbers should only contain numbers, and it should be 8 digits long";
```
Comment (amosting) [inline src/main/java/seedu/address/model/person/Phone.java:14]: add a fullstop

**Review by amosting — APPROVED** [review]
small nits otherwise LGTM!!

**Comment by ravern** [comment]
This can't be merged due to this being considered a feature enhancement right?
Can you close this PR if so?

**Comment by LHeng1** [comment]
Closing this PR since it cannot be fixed in our current iteration and can only be considered a feature enhancement.

**Pull request closed without merging**